src/unix/apple/disk.rs

```diff
@@ -72,44 +72,77 @@ impl DiskInner {
         self.is_read_only
     }
 
-    pub(crate) fn refresh(&mut self) -> bool {
-        #[cfg(target_os = "macos")]
-        let Some((read_bytes, written_bytes)) = self
-            .bsd_name
-            .as_ref()
-            .and_then(|name| crate::sys::inner::disk::get_disk_io(name))
-        else {
-            sysinfo_debug!("Failed to update disk i/o stats");
-            return false;
-        };
-        #[cfg(not(target_os = "macos"))]
-        let (read_bytes, written_bytes) = (0, 0);
+    pub(crate) fn refresh_specifics(&mut self, refresh_kind: DiskRefreshKind) -> bool {
+        if refresh_kind.kind() && self.type_ == DiskKind::Unknown(-1) {
+            let type_ = {
+                #[cfg(target_os = "macos")]
+                {
+                    self.bsd_name
+                        .as_ref()
+                        .and_then(|name| crate::sys::inner::disk::get_disk_type(name))
+                        .unwrap_or(DiskKind::Unknown(-1))
+                }
+                #[cfg(not(target_os = "macos"))]
+                DiskKind::SSD
+            };
 
-        self.old_read_bytes = self.read_bytes;
-        self.old_written_bytes = self.written_bytes;
-        self.read_bytes = read_bytes;
-        self.written_bytes = written_bytes;
+            self.type_ = type_;
+        }
 
-        unsafe {
-            if let Some(requested_properties) = build_requested_properties(&[
-                ffi::kCFURLVolumeAvailableCapacityKey,
-                ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey,
-            ]) {
-                match get_disk_properties(&self.volume_url, &requested_properties) {
-                    Some(disk_props) => {
-                        self.available_space = get_available_volume_space(&disk_props);
-                        true
-                    }
-                    None => {
-                        sysinfo_debug!("Failed to get disk properties");
-                        false
+        if refresh_kind.io_usage() {
+            let (read_bytes, written_bytes) = {
+                #[cfg(target_os = "macos")]
+                {
+                    self.bsd_name
+                        .as_ref()
+                        .and_then(|name| crate::sys::inner::disk::get_disk_io(name))
+                        .unwrap_or_else(|| {
+                            sysinfo_debug!("Failed to update disk i/o stats");
+                            (0, 0)
+                        })
+                }
+                #[cfg(not(target_os = "macos"))]
+                (0, 0)
+            };
+
+            self.old_read_bytes = self.read_bytes;
```

Comment:
Still not good here. It should not update the values if `get_disk_io` failed.

Reply:
Done